test/test_collections.py

```diff
@@ -339,5 +339,34 @@ def test_from_httplib_py2(self):
         self.assertEqual(d['www-authenticate'], 'asdf, bla')
         self.assertEqual(d.getlist('www-authenticate'), ['asdf', 'bla'])
 
+    def test_from_httplib_py2_invalid_first_header(self):
+        if six.PY3:
+            raise SkipTest("python3 has a different internal header implementation")
```

Comment:
We don't actually use this method on Python 3 at all - as far as I can tell, we're using it to mimic the Python 3 header data structure in Python 2.